dateconv/dateconv.go

```diff
@@ -134,6 +134,32 @@ func TotalDaysInBSYear(year int) (int, error) {
 	return sum, nil
 }
 
+// now returns the current time
+var now = func() time.Time {
```

Comment:
This seems like a hack if it's just to make testing easier.

Off the top of my head, an approach that makes sense is to create a private function `totalDaysSpannedUntilDate(d time.Time)`. The public function would just call this with the `now` value and we could just test the private function. Coverage slightly decreases but the tradeoff is worth it IMO.

**Brainstorming out loud**
I can see a more intricate solution by creating a `timer` interface. (bad name, but `time` is taken by the package)
```go
type timer interface {
  Now() time.Time
}
```

Then `ToBS()` would accept that as an argument. We could mock it out in tests by creating a fake `timer` impl.  It is more verbose though and requires us to have additional state to hold the `timer` interface in the first place. For a simple thing like this, the first approach should be fine.

Reply:
I found the hack to be very simple. I also thought of a better approach and read a few articles before proceeding with this approach.

Will update when I find the time.